test/integration-tests/src/test/java/org/wso2/testgrid/test/base/Utils.java

```diff
@@ -0,0 +1,197 @@
+     * @throws InterruptedException will be thrown at the Jar execution phase
+     * @throws  IOException be thrown at the Jar execution phase
+     */
+    public static int executeGenTestPlan(String[] args) throws InterruptedException, IOException {
+
+        //Constructing the execution command for generate test plans
+        String cmdCommand[] = { GEN_TEST_PLAN_SUB_COMMAND };
+        String cmdArgs[] = Stream.concat(Arrays.stream(cmdCommand), Arrays.stream(args)).toArray(String[]::new);
+
+        ShellExecutor executor = new ShellExecutor();
+        return executor.executeJar(getJarPath(), cmdArgs);
+
+    }
+
+    /**
+     * Executes the generate test report sub-command.
+     *
+     * @param args generate-test-plan sub-command related arguments e.g : --product
+     * @return Integer value with the execution status, 0 if success
+     * @throws InterruptedException will be thrown at the Jar execution phase
+     * @throws  IOException be thrown at the Jar execution phase
+     */
+    public static int executeGenReport(String[] args) throws IOException, InterruptedException {
+        String cmdCommand[] = { GEN_REPORT_SUB_COMMAND };
+        String cmdArgs[] = Stream.concat(Arrays.stream(cmdCommand), Arrays.stream(args)).toArray(String[]::new);
+
+        ShellExecutor executor = new ShellExecutor();
+        return executor.executeJar(getJarPath(), cmdArgs);
+    }
+
+    /**
+     * Construct the executable Jar absolute path and returns the value.
+     *
+     * @return Path of the executable Jar
+     */
+    private static String getJarPath() {
+        // Generating the actual Jar name with the version
+        String jarPath = TG_EXECUTABLE_LOCATION.replace("*", System.getProperty("project.version"));
+        return TESTGRID_UNZIP_LOCATION + jarPath;
+    }
+
+    /**
+     * Unzip a zip file into a given location.
+     *
+     * @param sourceFilePath - zip file need to extract
+     * @param extractedDir   - destination path given file to extract
+     * @throws IntegrationTestException is thrown if a exception occurs while unzipping the file
+     */
+    private static void extractFile(String sourceFilePath, String extractedDir) throws IntegrationTestException {
+        logger.info("Zip location is set to " + sourceFilePath + " || Unzip path is set to " + extractedDir);
+        FileOutputStream fileoutputstream = null;
+
+        String fileDestination = extractedDir + File.separator;
+        byte[] buf = new byte[1024];
+        ZipInputStream zipinputstream = null;
+        ZipEntry zipentry;
+        try {
+            zipinputstream = new ZipInputStream(new FileInputStream(sourceFilePath));
+
+            zipentry = zipinputstream.getNextEntry();
+
+            while (zipentry != null) {
+                //for each entry to be extracted
+                String entryName = fileDestination + zipentry.getName();
+                entryName = entryName.replace('/', File.separatorChar);
```

Comment:
we can have this on a single line (`entryName.replace('/', File.separatorChar).replace('\\', File.separatorChar)`)